### PR TITLE
Add S3 Browser Terraform

### LINKED ISSUE #139: As an end user, I want to be able to browse LROC data in S3 bucket
### Checked for duplicates

No - I haven't checked

### 🧑‍🔬 User Persona(s)

Data User

### 💪 Motivation

...so that I can [why do you want to do this?]

### 📖 Additional Details

_No response_

### Acceptance Criteria

**Given** an S3 bucket and prefix containing dataset files that are accessible through the configured browser interface
**When I perform** navigation to the browser URL mapped to that S3 bucket and prefix
**Then I expect** to see the S3 bucket contents displayed in the browser, so I can browse folders and access files under that prefix

### ⚙️ Engineering Details

_No response_

### 🎉 I&T

_No response_

### PR DESCRIPTION
**Pull request by eddiesarevalo**
## 🗒️ Summary
Adds terraform script to create s3-browser s3 bucket on aws.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [x] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [ ] AI generated substantial portions of this code

Estimated % of code influenced by AI: 20 %

## ⚙️ Test Data and/or Report
Follow readme.

## ♻️ Related Issues

Resolves #139 

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: terraform/backend-test.hcl [hunk 1]
```diff
@@ -0,0 +1,3 @@
+bucket = "pds-infra-test"
+key    = "dev/s3-browser.tfstate"
```
Comment (anilnatha) [inline terraform/backend-test.hcl]: This needs to be fixed to point to the "test" state path

File: terraform/variables.tf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+variable "managedby" {
+  description = "Tag value for owner managing the resource (E.g. for PDS Team we have PDS Team Email Distro)"
+  type        = string
+}
+
+variable "s3_bucket_name" {
+  description = "The name of the S3 bucket"
+  type        = string
+  default     = "my_test_bucket"
```
Comment (anilnatha) [inline terraform/variables.tf]: The S3 Bucket for S3 Browser should not have a default. We neeed to ensure we have set a name for it in our *.tfvars file by removing "default" to make this variable required.

File: src/js/bucketEndpoints.js
Comment (anilnatha) [inline src/js/bucketEndpoints.js:1]: I don't see the updated bucketEndpoints that we spoke of this morning.

File: src/js/bucketEndpoints.js [hunk 3]
```diff
@@ -62,6 +62,11 @@ export const bucketEndpoints = {
         "URL": "https://pds-css-archive.s3.us-west-2.amazonaws.com",
         "deepLinkPath": "sbn/gbo.ast.catalina.survey/",
         "appendPathToUrl": false
+    },
+    "lroc": {
+        "listingUrlPathPrefix": "data/store/img/",
+        "deepLinkPath": "lunar_reconnaissance_orbiter/pds4/lroc/lro-l-lroc-2-edr/",
+        "URL": "https://pds-sit.mcp.nasa.gov"
```
Comment (anilnatha) [inline src/js/bucketEndpoints.js]: This config will be an issue when we have to deploy for test and prod. Can we make this configurable in some way at build time?

**Review by anilnatha — APPROVED** [review]
Lookin' good!
